### PR TITLE
Convert to use class specific loggers (#428)

### LINKED ISSUE #428: Convert to class specific loggers
Currently there are a few loggers defined in base classes as protected which are then used by child/implementation classes. For example we might see:
```
08:20:50.031 [MainThread] ERROR org.eclipse.january.dataset.LazyDatasetBase - Could not parse datetime: 2021-04-20T08:20:50+01:00[Europe/London]
```
Where the message here is actually from `DateDatasetImpl` not `LazyDatasetBase`.

Using a logger for the class that the logger is actually used from increases clarity and saves time tracking down the code where the message originates.

### PR DESCRIPTION
**Pull request by joeshannon**
Replace use of inherited loggers with class specific logger fields.
Fixes #428

### CODE REVIEW

File: org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/Utils.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public void write(int b) {
 				}
 			}));
 	
-			LoggerFactory.getLogger(BasicExample.class);
+			LoggerFactory.getLogger(Utils.class);
```
Comment (joeshannon) [inline org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/Utils.java:45]: This wasn't strictly related but was matched by the procedure I used when checking for GDA classes using other classes' loggers.

File: org.eclipse.january/src/org/eclipse/january/dataset/DatasetUtils.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
 	/**
 	 * Setup the logging facilities
 	 */
-	transient protected static final Logger utilsLogger = LoggerFactory.getLogger(DatasetUtils.class);
+	transient private static final Logger utilsLogger = LoggerFactory.getLogger(DatasetUtils.class);
```
Comment (joeshannon) [inline org.eclipse.january/src/org/eclipse/january/dataset/DatasetUtils.java:35]: Not sure why this is transient but I guess is unrelated.

Comment (PeterC-DLS) [reply org.eclipse.january/src/org/eclipse/january/dataset/DatasetUtils.java:35]: You can remove the `transient` qualifier

File: org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/Utils.java (hunk 1 above)
Comment (PeterC-DLS) [reply org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/Utils.java:45]: Thanks

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**